tests/client/test_client_peer.py

```diff
@@ -111,58 +107,9 @@ async def test_init_client_peer_invalid_token(
     assert not peer_manager.peer_available("localhost")
 
 
-async def test_flow_client_peer(
```

Comment:
Maybe we can keep and replace `request_handler_protocol = protocol_factory()` with a protocol that does raw read/writes but that might be a bit complex to write a whole new one